cmd/storj-sim/network.go

```diff
@@ -25,8 +25,56 @@ import (
 	"storj.io/storj/internal/processgroup"
 )
 
+type peerClass int
+
+func (p peerClass) String() string {
+	return []string{"Satellite", "Version Control", "Boostrap", "Gateway", "Storage Node"}[p]
+}
+
+const (
+	// The following values of peer class, index, and endpoints are used
+	// to create a port with a consistent format for storj-sim services.
+
+	// Peer class
+	satellitePeer peerClass = iota
+	gatewayPeer
+	versioncontrolPeer
+	bootstrapPeer
+	storagenodePeer
+
+	// Index for peers with one instance (i.e. version control and bootstrap)
+	singleIndex = 0
+
+	// Endpoint
+	publicGRPC  = 0
+	privateGRPC = 1
+	publicHTTP  = 2
+	debugHTTP   = 9
+)
+
 const folderPermissions = 0744
 
+// port creates a port with a consistent format for storj-sim services.
+// The port format is: "1PXXE", where P is the peer class, XX is the index of the instance, and E is the endpoint.
+func port(peer peerClass, index, endpoint int) (string, error) {
+	const (
+		maxInstanceCount    = 100
+		maxStoragenodeCount = 200
+	)
+
+	switch {
```

Comment:
I would move this check into flag checking. That way you don't need to return an error and can keep rest of the code cleaner.

Reply:
good idea. moving this to the flag checking made this much simpler.